File: src/main/java/photon/libraries/PhotonLibrary.java
package photon.libraries;

import com.oracle.truffle.api.interop.InteropLibrary;
import com.oracle.truffle.api.library.ExportMessage;
import com.oracle.truffle.api.library.GenerateLibrary;
import com.oracle.truffle.api.library.Library;

@GenerateLibrary
@GenerateLibrary.DefaultExport(IntegerMethods.class)
public abstract class PhotonLibrary extends Library {
    public Object invokeMember(Object receiver, String member, Object... arguments) {
        return false;
    }
}

File: src/main/kotlin/photon/compiler/libraries/PhotonLibrary.kt
package photon.compiler.libraries

import com.oracle.truffle.api.interop.*
import com.oracle.truffle.api.library.*
import photon.compiler.core.EvalMode

@GenerateLibrary
abstract class PhotonLibrary: Library() {
  @Throws(UnknownIdentifierException::class)
  open fun invokeMember(receiver: Any, evalMode: EvalMode, member: String, vararg arguments: Any): Any {
    throw UnsupportedMessageException.create()
  }
}
